internal/security/secretstoreclient/vault.go

```diff
@@ -97,68 +63,33 @@ func (vc *vaultClient) Init(config SecretServiceInfo, vmkWriter io.Writer) (stat
 		SecretShares:    config.VaultSecretShares,
 		SecretThreshold: config.VaultSecretThreshold,
 	}
-
-	vc.logger.Info(fmt.Sprintf("vault init strategy (SSS parameters): shares=%d threshold=%d", initRequest.SecretShares, initRequest.SecretThreshold))
-
-	url := vc.buildURL(VaultInitAPI)
-	resp, err := vc.commonRequest(nil, http.MethodPost, url, &initRequest)
-	if err != nil {
-		vc.logger.Error(fmt.Sprintf("failed to send Vault init request: %s", err.Error()))
-		return 0, err
-	}
-	defer resp.Body.Close()
-
-	if resp.StatusCode != http.StatusOK {
-		vc.logger.Error(fmt.Sprintf("vault init request failed with status: %s", resp.Status))
-		return resp.StatusCode, err
-	}
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		vc.logger.Error(fmt.Sprintf("failed to fetch the Vault init request response body: %s", err.Error()))
-		return 0, err
-	}
-
 	initResp := InitResponse{}
-	if err = json.Unmarshal(body, &initResp); err != nil {
-		vc.logger.Error(fmt.Sprintf("failed to build the JSON structure from the init request response body: %s", err.Error()))
-		return 0, err
-	}
 
-	_, err = vmkWriter.Write(body)
-
-	if err != nil {
-		vc.logger.Error(fmt.Sprintf("failed to create Vault init response %s file, HTTP status: %s", config.TokenFolderPath+"/"+config.TokenFile, err.Error()))
-		return 0, err
-	}
+	vc.logger.Info(fmt.Sprintf("vault init strategy (SSS parameters): shares=%d threshold=%d", initRequest.SecretShares, initRequest.SecretThreshold))
 
-	vc.logger.Info("Vault initialization complete.")
-	return resp.StatusCode, nil
+	resp, err := vc.commonJSONRequest(nil, http.MethodPost, VaultInitAPI, &initRequest)
+	return vc.processResponse(resp, err, "initialize secret store", http.StatusOK, &initResp, func() error {
```

Comment:
I think it's more readable code to just do:

```go
	err = vc.processResponse(resp, err, "initialize secret store", http.StatusOK, &initResp)
	if err != nil {
	    return err
	}
	return json.NewEncoder(vmkWriter).Encode(initResp)
}
```

you're only saving 3 lines of code checking the error here and IMHO it's more confusing to use this `processResponse` function since you are _requiring_ it to essentially be a closure over some other variable that's probably already provided to the `processResponse` function, so all in all this ends up feeling quite weird to read through.

Reply:
I see what you are saying.  In the delegate flow, it will do the json processing then log the success info line.  In this example, processResponse() would log "successfully made request to intialize secret store" or log an error, then do the postprocessing.

Reply:
Addressed in #1855